File: 백준/Silver/2852. NBA 농구/NBA 농구.py
game_time = 60*48
score = [0, 0, 0]
times = [0, 0, 0]

N = int(input())

for _ in range (N):
    team, time = map(str,input().split())
    min, sec = map(int, time.split(":"))

    time = min*60 + sec
    team = int(team)

    if team == 1:
        score[team] +=1
        if score[1] > score[2] and score[1] -1 == score[2] :
            times[1] += game_time - time
        elif score[1] == score[2]:
            times[2] = (times[2] - (game_time - time))

    elif team == 2:
        score[team] += 1
        if score[1] < score[2] and score[2] -1 == score[1]:
            times[2] += game_time - time
        elif score[1] == score[2]:
            times[1] = (times[1] - (game_time - time)) 

print("{:02d}:{:02d}".format(int(times[1] / 60), times[1] % 60))
print("{:02d}:{:02d}".format(int(times[2] / 60), times[2] % 60))

File: 백준/Silver/2852. NBA 농구/NBA 농구.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.PriorityQueue;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		
		int N = Integer.parseInt(br.readLine());
		int times[] = new int[2881];

		for (int i = 0; i < N; i++) {
			st = new StringTokenizer(br.readLine());
			int teamNo = Integer.parseInt(st.nextToken());
			String[] strTime = st.nextToken().split(":");

			int hour = Integer.parseInt(strTime[0]);
			int minute = Integer.parseInt(strTime[1]);

			int goalTime = hour * 60 + minute;

			times[goalTime] = teamNo;
		}

		solve(times);
	}
	
	private static void solve(int[] times) {

		int team1GoalCnt = 0;
		int team2GaolCnt = 0;
		int team1Timer = 0;
		int team2Timer = 0;

		int time = 0;

		while (time <= 2880) {

			if (team1GoalCnt > team2GaolCnt) {
				team1Timer ++;
			}

			if (team1GoalCnt < team2GaolCnt) {
				team2Timer ++;
			}


			if (times[time] != 0) {
				if (times[time] == 1) {
					team1GoalCnt++;
				} else {
					team2GaolCnt++;
				}
			}

			time++;
		}

		getTimeFormat(team1Timer);
		getTimeFormat(team2Timer);
	}

	private static void getTimeFormat(int timer) {

		int hour = timer / 60;
		int min = timer % 60;

		System.out.printf("%02d:%02d%n", hour, min);
	}
}
